\documentclass[nofonts,]{tufte-handout}

% ams
\usepackage{amssymb,amsmath}

\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \makeatletter
  \@ifpackageloaded{fontspec}{}{\usepackage{fontspec}}
  \makeatother
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
  \makeatletter
  \@ifpackageloaded{soul}{
     \renewcommand\allcapsspacing[1]{{\addfontfeature{LetterSpace=15}#1}}
     \renewcommand\smallcapsspacing[1]{{\addfontfeature{LetterSpace=10}#1}}
   }{}
  \makeatother
\fi

% graphix
\usepackage{graphicx}
\setkeys{Gin}{width=\linewidth,totalheight=\textheight,keepaspectratio}

% booktabs
\usepackage{booktabs}

% url
\usepackage{url}

% hyperref
\usepackage{hyperref}

% units.
\usepackage{units}


\setcounter{secnumdepth}{-1}

% citations
\usepackage{natbib}
\bibliographystyle{plainnat}

%% tint override
\setcitestyle{round} 

% pandoc syntax highlighting
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}

% longtable

% multiplecol
\usepackage{multicol}

% strikeout
\usepackage[normalem]{ulem}

% morefloats
\usepackage{morefloats}


% tightlist macro required by pandoc >= 1.14
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}

% title / author / date
\title{UCSCXenaTools: Download Public Cancer Genomic Data from UCSC Xena Hubs}
\author{Shixiang Wang}
\date{2019-06-17}

%% -- tint overrides
%% fonts, using roboto (condensed) as default
\usepackage[sfdefault,condensed]{roboto}
%% also nice: \usepackage[default]{lato}

%% colored links, setting 'borrowed' from RJournal.sty with 'Thanks, Achim!'
\RequirePackage{color}
\definecolor{link}{rgb}{0.1,0.1,0.8} %% blue with some grey
\hypersetup{
  colorlinks,%
  citecolor=link,%
  filecolor=link,%
  linkcolor=link,%
  urlcolor=link
}

%% macros
\makeatletter

%% -- tint does not use italics or allcaps in title
\renewcommand{\maketitle}{%     
  \newpage
  \global\@topnum\z@% prevent floats from being placed at the top of the page
  \begingroup
    \setlength{\parindent}{0pt}%
    \setlength{\parskip}{4pt}%
    \let\@@title\@empty
    \let\@@author\@empty
    \let\@@date\@empty
    \ifthenelse{\boolean{@tufte@sfsidenotes}}{%
      %\gdef\@@title{\sffamily\LARGE\allcaps{\@title}\par}%
      %\gdef\@@author{\sffamily\Large\allcaps{\@author}\par}%
      %\gdef\@@date{\sffamily\Large\allcaps{\@date}\par}%
      \gdef\@@title{\begingroup\fontseries{b}\selectfont\LARGE{\@title}\par}%
      \gdef\@@author{\begingroup\fontseries{l}\selectfont\Large{\@author}\par}%
      \gdef\@@date{\begingroup\fontseries{l}\selectfont\Large{\@date}\par}%
    }{%
      %\gdef\@@title{\LARGE\itshape\@title\par}%
      %\gdef\@@author{\Large\itshape\@author\par}%
      %\gdef\@@date{\Large\itshape\@date\par}%
      \gdef\@@title{\begingroup\fontseries{b}\selectfont\LARGE\@title\par\endgroup}%
      \gdef\@@author{\begingroup\fontseries{l}\selectfont\Large\@author\par\endgroup}%
      \gdef\@@date{\begingroup\fontseries{l}\selectfont\Large\@date\par\endgroup}%
    }%
    \@@title
    \@@author
    \@@date
  \endgroup
  \thispagestyle{plain}% suppress the running head
  \tuftebreak% add some space before the text begins
  \@afterindentfalse\@afterheading% suppress indentation of the next paragraph
}

%% -- tint does not use italics or allcaps in section/subsection/paragraph
\titleformat{\section}%
  [hang]% shape
  %{\normalfont\Large\itshape}% format applied to label+text
  {\fontseries{b}\selectfont\Large}% format applied to label+text
  {\thesection}% label
  {1em}% horizontal separation between label and title body
  {}% before the title body
  []% after the title body

\titleformat{\subsection}%
  [hang]% shape
  %{\normalfont\large\itshape}% format applied to label+text
  {\fontseries{m}\selectfont\large}% format applied to label+text
  {\thesubsection}% label
  {1em}% horizontal separation between label and title body
  {}% before the title body
  []% after the title body

\titleformat{\paragraph}%
  [runin]% shape
  %{\normalfont\itshape}% format applied to label+text
  {\fontseries{l}\selectfont}% format applied to label+text
  {\theparagraph}% label
  {1em}% horizontal separation between label and title body
  {}% before the title body
  []% after the title body

%% -- tint does not use italics here either
% Formatting for main TOC (printed in front matter)
% {section} [left] {above} {before w/label} {before w/o label} {filler + page} [after]
\ifthenelse{\boolean{@tufte@toc}}{%
  \titlecontents{part}% FIXME
    [0em] % distance from left margin
    %{\vspace{1.5\baselineskip}\begin{fullwidth}\LARGE\rmfamily\itshape} % above (global formatting of entry)
    {\vspace{1.5\baselineskip}\begin{fullwidth}\fontseries{m}\selectfont\LARGE} % above (global formatting of entry)
    {\contentslabel{2em}} % before w/label (label = ``II'')
    {} % before w/o label
    {\rmfamily\upshape\qquad\thecontentspage} % filler + page (leaders and page num)
    [\end{fullwidth}] % after
  \titlecontents{chapter}%
    [0em] % distance from left margin
    %{\vspace{1.5\baselineskip}\begin{fullwidth}\LARGE\rmfamily\itshape} % above (global formatting of entry)
    {\vspace{1.5\baselineskip}\begin{fullwidth}\fontseries{m}\selectfont\LARGE} % above (global formatting of entry)
    {\hspace*{0em}\contentslabel{2em}} % before w/label (label = ``2'')
    {\hspace*{0em}} % before w/o label
    %{\rmfamily\upshape\qquad\thecontentspage} % filler + page (leaders and page num)
    {\upshape\qquad\thecontentspage} % filler + page (leaders and page num)
    [\end{fullwidth}] % after
  \titlecontents{section}% FIXME
    [0em] % distance from left margin
    %{\vspace{0\baselineskip}\begin{fullwidth}\Large\rmfamily\itshape} % above (global formatting of entry)
    {\vspace{0\baselineskip}\begin{fullwidth}\fontseries{m}\selectfont\Large} % above (global formatting of entry)
    {\hspace*{2em}\contentslabel{2em}} % before w/label (label = ``2.6'')
    {\hspace*{2em}} % before w/o label
    %{\rmfamily\upshape\qquad\thecontentspage} % filler + page (leaders and page num)
    {\upshape\qquad\thecontentspage} % filler + page (leaders and page num)
    [\end{fullwidth}] % after
  \titlecontents{subsection}% FIXME
    [0em] % distance from left margin
    %{\vspace{0\baselineskip}\begin{fullwidth}\large\rmfamily\itshape} % above (global formatting of entry)
    {\vspace{0\baselineskip}\begin{fullwidth}\fontseries{m}\selectfont\large} % above (global formatting of entry)
    {\hspace*{4em}\contentslabel{4em}} % before w/label (label = ``2.6.1'')
    {\hspace*{4em}} % before w/o label
    %{\rmfamily\upshape\qquad\thecontentspage} % filler + page (leaders and page num)
    {\upshape\qquad\thecontentspage} % filler + page (leaders and page num)
    [\end{fullwidth}] % after
  \titlecontents{paragraph}% FIXME
    [0em] % distance from left margin
    %{\vspace{0\baselineskip}\begin{fullwidth}\normalsize\rmfamily\itshape} % above (global formatting of entry)
    {\vspace{0\baselineskip}\begin{fullwidth}\fontseries{m}\selectfont\normalsize\rmfamily} % above (global formatting of entry)
    {\hspace*{6em}\contentslabel{2em}} % before w/label (label = ``2.6.0.0.1'')
    {\hspace*{6em}} % before w/o label
    %{\rmfamily\upshape\qquad\thecontentspage} % filler + page (leaders and page num)
    {\upshape\qquad\thecontentspage} % filler + page (leaders and page num)
    [\end{fullwidth}] % after
}{}

  
\makeatother



\begin{document}

\maketitle




\textbf{UCSCXenaTools} is an R package for downloading and exploring
data from \textbf{UCSC Xena data hubs}, which are

\begin{quote}
a collection of UCSC-hosted public databases such as TCGA, ICGC, TARGET,
GTEx, CCLE, and others. Databases are normalized so they can be
combined, linked, filtered, explored and downloaded.

-- \href{https://xena.ucsc.edu/}{UCSC Xena}
\end{quote}

If you use this package in academic field, please cite:

\begin{verbatim}
Wang, Shixiang, et al. "The predictive power of tumor mutational burden 
    in lung cancer immunotherapy response is influenced by patients' sex." 
    International journal of cancer (2019).
\end{verbatim}

\hypertarget{installation}{%
\section{Installation}\label{installation}}

Install stable release from CRAN with:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{install.packages}\NormalTok{(}\StringTok{"UCSCXenaTools"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

You can also install devel version of \textbf{UCSCXenaTools} from github
with:

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# install.packages('remotes')}
\NormalTok{remotes}\OperatorTok{::}\KeywordTok{install_github}\NormalTok{(}\StringTok{"ShixiangWang/UCSCXenaTools"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Read this vignettes.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{browseVignettes}\NormalTok{(}\StringTok{"UCSCXenaTools"}\NormalTok{)}
\CommentTok{# or}
\StringTok{`}\DataTypeTok{?}\StringTok{`}\NormalTok{(}\StringTok{`}\DataTypeTok{?}\StringTok{`}\NormalTok{(UCSCXenaTools))}
\end{Highlighting}
\end{Shaded}

\hypertarget{data-hub-list}{%
\section{Data Hub List}\label{data-hub-list}}

All datasets are available at \url{https://xenabrowser.net/datapages/}.

Currently, \textbf{UCSCXenaTools} supports 10 data hubs of UCSC Xena.

\begin{itemize}
\tightlist
\item
  UCSC Public Hub: \url{https://ucscpublic.xenahubs.net}
\item
  TCGA Hub: \url{https://tcga.xenahubs.net}
\item
  GDC Xena Hub: \url{https://gdc.xenahubs.net}
\item
  ICGC Xena Hub: \url{https://icgc.xenahubs.net}
\item
  Pan-Cancer Atlas Hub: \url{https://pancanatlas.xenahubs.net}
\item
  GA4GH (TOIL) Hub: \url{https://toil.xenahubs.net}
\item
  Treehouse Hub: \url{https://xena.treehouse.gi.ucsc.edu}
\item
  PCAWG Hub: \url{https://pcawg.xenahubs.net}
\item
  ATAC-seq Hub: \url{https://atacseq.xenahubs.net}
\item
  Singel Cell Xena hub: \url{https://singlecell.xenahubs.net}
\end{itemize}

If any url of data hubs are changed or a new data hub is online, please
remind me by emailing to
\href{mailto:w_shixiang@163.com}{\nolinkurl{w\_shixiang@163.com}} or
\href{https://github.com/ShixiangWang/UCSCXenaTools/issues}{opening an
issue on GitHub}.

\hypertarget{usage}{%
\section{Usage}\label{usage}}

Download UCSC Xena Datasets and load them into R by
\textbf{UCSCXenaTools} is a workflow with \texttt{generate},
\texttt{filter}, \texttt{query}, \texttt{download} and \texttt{prepare}
5 steps, which are implemented as \texttt{XenaGenerate},
\texttt{XenaFilter}, \texttt{XenaQuery}, \texttt{XenaDownload} and
\texttt{XenaPrepare} functions, respectively. They are very clear and
easy to use and combine with other packages like \texttt{dplyr}.

To show the basic usage of \textbf{UCSCXenaTools}, we will download
clinical data of LUNG, LUAD, LUSC from TCGA (hg19 version) data hub.

\hypertarget{xenadata-data.frame}{%
\subsection{XenaData data.frame}\label{xenadata-data.frame}}

Begin from version \texttt{0.2.0}, \textbf{UCSCXenaTools} uses a
\texttt{data.frame} object (built in package, someone may call it
\texttt{tibble}) \texttt{XenaData} to generate an instance of
\texttt{XenaHub} class to record general information of all datasets of
UCSC Xena Data Hubs.

You can load \texttt{XenaData} after loading \texttt{UCSCXenaTools} into
R.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(UCSCXenaTools)}
\CommentTok{#> =========================================================================}
\CommentTok{#> UCSCXenaTools version 1.2.2}
\CommentTok{#> Github page: https://github.com/ShixiangWang/UCSCXenaTools}
\CommentTok{#> Documentation: https://shixiangwang.github.io/UCSCXenaTools/}
\CommentTok{#> }
\CommentTok{#> If you use it in published research, please cite:}
\CommentTok{#> Wang, Shixiang, et al. "The predictive power of tumor mutational burden}
\CommentTok{#>     in lung cancer immunotherapy response is influenced by patients' sex."}
\CommentTok{#>     International journal of cancer (2019).}
\CommentTok{#> =========================================================================}
\CommentTok{#> }
\KeywordTok{data}\NormalTok{(XenaData)}

\KeywordTok{head}\NormalTok{(XenaData)}
\CommentTok{#> # A tibble: 6 x 17}
\CommentTok{#>   XenaHosts XenaHostNames XenaCohorts}
\CommentTok{#>   <chr>     <chr>         <chr>      }
\CommentTok{#> 1 https://~ publicHub     Acute lymp~}
\CommentTok{#> 2 https://~ publicHub     Acute lymp~}
\CommentTok{#> 3 https://~ publicHub     Acute lymp~}
\CommentTok{#> 4 https://~ publicHub     Breast Can~}
\CommentTok{#> 5 https://~ publicHub     Breast Can~}
\CommentTok{#> 6 https://~ publicHub     Breast Can~}
\CommentTok{#> # ... with 14 more variables:}
\CommentTok{#> #   XenaDatasets <chr>, SampleCount <chr>,}
\CommentTok{#> #   DataSubtype <chr>, Label <chr>,}
\CommentTok{#> #   Type <chr>, AnatomicalOrigin <chr>,}
\CommentTok{#> #   SampleType <chr>, Tags <chr>,}
\CommentTok{#> #   ProbeMap <chr>, LongTitle <chr>,}
\CommentTok{#> #   Citation <chr>, Version <chr>,}
\CommentTok{#> #   Unit <chr>, Platform <chr>}
\end{Highlighting}
\end{Shaded}

Names of all hub names/urls can be accessed by object
\texttt{.xena\_hosts}:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{UCSCXenaTools}\OperatorTok{:::}\NormalTok{.xena_hosts}
\CommentTok{#>    https://ucscpublic.xenahubs.net }
\CommentTok{#>                        "publicHub" }
\CommentTok{#>          https://tcga.xenahubs.net }
\CommentTok{#>                          "tcgaHub" }
\CommentTok{#>           https://gdc.xenahubs.net }
\CommentTok{#>                           "gdcHub" }
\CommentTok{#>          https://icgc.xenahubs.net }
\CommentTok{#>                          "icgcHub" }
\CommentTok{#>          https://toil.xenahubs.net }
\CommentTok{#>                          "toilHub" }
\CommentTok{#>   https://pancanatlas.xenahubs.net }
\CommentTok{#>                   "pancanAtlasHub" }
\CommentTok{#> https://xena.treehouse.gi.ucsc.edu }
\CommentTok{#>                     "treehouseHub" }
\CommentTok{#>         https://pcawg.xenahubs.net }
\CommentTok{#>                         "pcawgHub" }
\CommentTok{#>       https://atacseq.xenahubs.net }
\CommentTok{#>                       "atacseqHub" }
\CommentTok{#>    https://singlecell.xenahubs.net }
\CommentTok{#>                    "singlecellHub"}
\end{Highlighting}
\end{Shaded}

\hypertarget{generate-a-xenahub-object}{%
\subsection{Generate a XenaHub object}\label{generate-a-xenahub-object}}

This can be implemented by \texttt{XenaGenerate} function, which
generates \texttt{XenaHub} object from \texttt{XenaData} data frame.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{XenaGenerate}\NormalTok{()}
\CommentTok{#> class: XenaHub }
\CommentTok{#> hosts():}
\CommentTok{#>   https://ucscpublic.xenahubs.net}
\CommentTok{#>   https://tcga.xenahubs.net}
\CommentTok{#>   https://gdc.xenahubs.net}
\CommentTok{#>   https://icgc.xenahubs.net}
\CommentTok{#>   https://toil.xenahubs.net}
\CommentTok{#>   https://pancanatlas.xenahubs.net}
\CommentTok{#>   https://xena.treehouse.gi.ucsc.edu}
\CommentTok{#>   https://pcawg.xenahubs.net}
\CommentTok{#>   https://atacseq.xenahubs.net}
\CommentTok{#>   https://singlecell.xenahubs.net}
\CommentTok{#> cohorts() (140 total):}
\CommentTok{#>   Acute lymphoblastic leukemia (Mullighan 2008)}
\CommentTok{#>   Breast Cancer (Caldas 2007)}
\CommentTok{#>   Breast Cancer (Chin 2006)}
\CommentTok{#>   ...}
\CommentTok{#>   human brain transcriptome (Darmanis PNAS 2015)}
\CommentTok{#>   mouse cortex and hippocampus (Zeisel Science 2015)}
\CommentTok{#> datasets() (1646 total):}
\CommentTok{#>   mullighan2008_public/mullighan2008_500K_genomicMatrix}
\CommentTok{#>   mullighan2008_public/mullighan2008_public_clinicalMatrix}
\CommentTok{#>   mullighan2008_public/mullighan2008_SNP6_genomicMatrix}
\CommentTok{#>   ...}
\CommentTok{#>   Zeisel/Zeisel_expression_mRNA_log2}
\CommentTok{#>   Zeisel/Zeisel_expression_phenotype}
\end{Highlighting}
\end{Shaded}

You can set \texttt{subset} argument to narrow datasets.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{XenaGenerate}\NormalTok{(}\DataTypeTok{subset =}\NormalTok{ XenaHostNames }\OperatorTok{==}\StringTok{ "tcgaHub"}\NormalTok{)}
\CommentTok{#> class: XenaHub }
\CommentTok{#> hosts():}
\CommentTok{#>   https://tcga.xenahubs.net}
\CommentTok{#> cohorts() (38 total):}
\CommentTok{#>   TCGA Acute Myeloid Leukemia (LAML)}
\CommentTok{#>   TCGA Adrenocortical Cancer (ACC)}
\CommentTok{#>   TCGA Bile Duct Cancer (CHOL)}
\CommentTok{#>   ...}
\CommentTok{#>   TCGA Thyroid Cancer (THCA)}
\CommentTok{#>   TCGA Uterine Carcinosarcoma (UCS)}
\CommentTok{#> datasets() (879 total):}
\CommentTok{#>   TCGA.LAML.sampleMap/HumanMethylation27}
\CommentTok{#>   TCGA.LAML.sampleMap/HumanMethylation450}
\CommentTok{#>   TCGA.LAML.sampleMap/Gistic2_CopyNumber_Gistic2_all_data_by_genes}
\CommentTok{#>   ...}
\CommentTok{#>   TCGA.UCS.sampleMap/Pathway_Paradigm_RNASeq_And_Copy_Number}
\CommentTok{#>   TCGA.UCS.sampleMap/mutation_curated_broad}
\end{Highlighting}
\end{Shaded}

\begin{quote}
You can also use \texttt{XenaHub()} to generate a \texttt{XenaHub}
object for API communication, but it is not recommended.
\end{quote}

It's possible to extract info from \texttt{XenaHub} object by
\texttt{hosts()}, \texttt{cohorts()} and \texttt{datasets()}.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{xe =}\StringTok{ }\KeywordTok{XenaGenerate}\NormalTok{(}\DataTypeTok{subset =}\NormalTok{ XenaHostNames }\OperatorTok{==}\StringTok{ "tcgaHub"}\NormalTok{)}
\CommentTok{# get hosts}
\KeywordTok{hosts}\NormalTok{(xe)}
\CommentTok{#> [1] "https://tcga.xenahubs.net"}
\CommentTok{# get cohorts}
\KeywordTok{head}\NormalTok{(}\KeywordTok{cohorts}\NormalTok{(xe))}
\CommentTok{#> [1] "TCGA Acute Myeloid Leukemia (LAML)"}
\CommentTok{#> [2] "TCGA Adrenocortical Cancer (ACC)"  }
\CommentTok{#> [3] "TCGA Bile Duct Cancer (CHOL)"      }
\CommentTok{#> [4] "TCGA Bladder Cancer (BLCA)"        }
\CommentTok{#> [5] "TCGA Breast Cancer (BRCA)"         }
\CommentTok{#> [6] "TCGA Cervical Cancer (CESC)"}
\CommentTok{# get datasets}
\KeywordTok{head}\NormalTok{(}\KeywordTok{datasets}\NormalTok{(xe))}
\CommentTok{#> [1] "TCGA.LAML.sampleMap/HumanMethylation27"                          }
\CommentTok{#> [2] "TCGA.LAML.sampleMap/HumanMethylation450"                         }
\CommentTok{#> [3] "TCGA.LAML.sampleMap/Gistic2_CopyNumber_Gistic2_all_data_by_genes"}
\CommentTok{#> [4] "TCGA.LAML.sampleMap/mutation_wustl_hiseq"                        }
\CommentTok{#> [5] "TCGA.LAML.sampleMap/GA"                                          }
\CommentTok{#> [6] "TCGA.LAML.sampleMap/HiSeqV2_percentile"}
\end{Highlighting}
\end{Shaded}

Pipe operator \texttt{\%\textgreater{}\%} can also be used here.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(dplyr)}
\NormalTok{XenaData }\OperatorTok{%>%}\StringTok{ }\KeywordTok{filter}\NormalTok{(XenaHostNames }\OperatorTok{==}\StringTok{ "tcgaHub"}\NormalTok{, }
    \KeywordTok{grepl}\NormalTok{(}\StringTok{"BRCA"}\NormalTok{, XenaCohorts), }\KeywordTok{grepl}\NormalTok{(}\StringTok{"Path"}\NormalTok{, }
\NormalTok{        XenaDatasets)) }\OperatorTok{%>%}\StringTok{ }\KeywordTok{XenaGenerate}\NormalTok{()}
\CommentTok{#> class: XenaHub }
\CommentTok{#> hosts():}
\CommentTok{#>   https://tcga.xenahubs.net}
\CommentTok{#> cohorts() (1 total):}
\CommentTok{#>   TCGA Breast Cancer (BRCA)}
\CommentTok{#> datasets() (4 total):}
\CommentTok{#>   TCGA.BRCA.sampleMap/Pathway_Paradigm_mRNA_And_Copy_Number}
\CommentTok{#>   TCGA.BRCA.sampleMap/Pathway_Paradigm_RNASeq}
\CommentTok{#>   TCGA.BRCA.sampleMap/Pathway_Paradigm_RNASeq_And_Copy_Number}
\CommentTok{#>   TCGA.BRCA.sampleMap/Pathway_Paradigm_mRNA}
\end{Highlighting}
\end{Shaded}

\hypertarget{filter}{%
\subsection{Filter}\label{filter}}

There are too many datasets in \texttt{xe}, you can filter them by
\texttt{XenaFilter} function. Regular expression can be used here.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{(xe2 <-}\StringTok{ }\KeywordTok{XenaFilter}\NormalTok{(xe, }\DataTypeTok{filterDatasets =} \StringTok{"clinical"}\NormalTok{))}
\CommentTok{#> class: XenaHub }
\CommentTok{#> hosts():}
\CommentTok{#>   https://tcga.xenahubs.net}
\CommentTok{#> cohorts() (37 total):}
\CommentTok{#>   TCGA Acute Myeloid Leukemia (LAML)}
\CommentTok{#>   TCGA Adrenocortical Cancer (ACC)}
\CommentTok{#>   TCGA Bile Duct Cancer (CHOL)}
\CommentTok{#>   ...}
\CommentTok{#>   TCGA Thyroid Cancer (THCA)}
\CommentTok{#>   TCGA Uterine Carcinosarcoma (UCS)}
\CommentTok{#> datasets() (37 total):}
\CommentTok{#>   TCGA.LAML.sampleMap/LAML_clinicalMatrix}
\CommentTok{#>   TCGA.ACC.sampleMap/ACC_clinicalMatrix}
\CommentTok{#>   TCGA.CHOL.sampleMap/CHOL_clinicalMatrix}
\CommentTok{#>   ...}
\CommentTok{#>   TCGA.THCA.sampleMap/THCA_clinicalMatrix}
\CommentTok{#>   TCGA.UCS.sampleMap/UCS_clinicalMatrix}
\end{Highlighting}
\end{Shaded}

Then select \texttt{LUAD}, \texttt{LUSC} and \texttt{LUNG} 3 datasets.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{xe2 <-}\StringTok{ }\KeywordTok{XenaFilter}\NormalTok{(xe2, }\DataTypeTok{filterDatasets =} \StringTok{"LUAD|LUSC|LUNG"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Pipe can be used here.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{xe }\OperatorTok{%>%}\StringTok{ }\KeywordTok{XenaFilter}\NormalTok{(}\DataTypeTok{filterDatasets =} \StringTok{"clinical"}\NormalTok{) }\OperatorTok{%>%}\StringTok{ }
\StringTok{    }\KeywordTok{XenaFilter}\NormalTok{(}\DataTypeTok{filterDatasets =} \StringTok{"luad|lusc|lung"}\NormalTok{)}
\CommentTok{#> class: XenaHub }
\CommentTok{#> hosts():}
\CommentTok{#>   https://tcga.xenahubs.net}
\CommentTok{#> cohorts() (3 total):}
\CommentTok{#>   TCGA Lung Adenocarcinoma (LUAD)}
\CommentTok{#>   TCGA Lung Cancer (LUNG)}
\CommentTok{#>   TCGA Lung Squamous Cell Carcinoma (LUSC)}
\CommentTok{#> datasets() (3 total):}
\CommentTok{#>   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix}
\CommentTok{#>   TCGA.LUNG.sampleMap/LUNG_clinicalMatrix}
\CommentTok{#>   TCGA.LUSC.sampleMap/LUSC_clinicalMatrix}
\end{Highlighting}
\end{Shaded}

\hypertarget{browse-datasets}{%
\subsection{Browse datasets}\label{browse-datasets}}

Sometimes, you may want to check data before you query and download
data. A new feature \texttt{XenaBrowse} is implemented in
\textbf{UCSCXenaTools}.

Create two XenaHub objects:

\begin{itemize}
\tightlist
\item
  \texttt{to\_browse} - a XenaHub object contains a cohort and a
  dataset.
\item
  \texttt{to\_browse2} - a XenaHub object contains 2 cohorts and 2
  datasets.
\end{itemize}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{to_browse <-}\StringTok{ }\KeywordTok{XenaGenerate}\NormalTok{(}\DataTypeTok{subset =}\NormalTok{ XenaHostNames }\OperatorTok{==}\StringTok{ }
\StringTok{    "tcgaHub"}\NormalTok{) }\OperatorTok{%>%}\StringTok{ }\KeywordTok{XenaFilter}\NormalTok{(}\DataTypeTok{filterDatasets =} \StringTok{"clinical"}\NormalTok{) }\OperatorTok{%>%}\StringTok{ }
\StringTok{    }\KeywordTok{XenaFilter}\NormalTok{(}\DataTypeTok{filterDatasets =} \StringTok{"LUAD"}\NormalTok{)}

\NormalTok{to_browse}
\CommentTok{#> class: XenaHub }
\CommentTok{#> hosts():}
\CommentTok{#>   https://tcga.xenahubs.net}
\CommentTok{#> cohorts() (1 total):}
\CommentTok{#>   TCGA Lung Adenocarcinoma (LUAD)}
\CommentTok{#> datasets() (1 total):}
\CommentTok{#>   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix}

\NormalTok{to_browse2 <-}\StringTok{ }\KeywordTok{XenaGenerate}\NormalTok{(}\DataTypeTok{subset =}\NormalTok{ XenaHostNames }\OperatorTok{==}\StringTok{ }
\StringTok{    "tcgaHub"}\NormalTok{) }\OperatorTok{%>%}\StringTok{ }\KeywordTok{XenaFilter}\NormalTok{(}\DataTypeTok{filterDatasets =} \StringTok{"clinical"}\NormalTok{) }\OperatorTok{%>%}\StringTok{ }
\StringTok{    }\KeywordTok{XenaFilter}\NormalTok{(}\DataTypeTok{filterDatasets =} \StringTok{"LUAD|LUSC"}\NormalTok{)}

\NormalTok{to_browse2}
\CommentTok{#> class: XenaHub }
\CommentTok{#> hosts():}
\CommentTok{#>   https://tcga.xenahubs.net}
\CommentTok{#> cohorts() (2 total):}
\CommentTok{#>   TCGA Lung Adenocarcinoma (LUAD)}
\CommentTok{#>   TCGA Lung Squamous Cell Carcinoma (LUSC)}
\CommentTok{#> datasets() (2 total):}
\CommentTok{#>   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix}
\CommentTok{#>   TCGA.LUSC.sampleMap/LUSC_clinicalMatrix}
\end{Highlighting}
\end{Shaded}

\texttt{XenaBrowse()} function can be used to browse dataset/cohort
links using your default web browser. At default, this function limit
one dataset/cohort for preventing user to open too many links at once.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# This will open you web browser}
\KeywordTok{XenaBrowse}\NormalTok{(to_browse)}

\KeywordTok{XenaBrowse}\NormalTok{(to_browse, }\DataTypeTok{type =} \StringTok{"cohort"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# This will throw error}
\KeywordTok{XenaBrowse}\NormalTok{(to_browse2)}
\CommentTok{#> Error in XenaBrowse(to_browse2): This function limite 1 dataset to browse.}
\CommentTok{#>  Set multiple to TRUE if you want to browse multiple links.}

\KeywordTok{XenaBrowse}\NormalTok{(to_browse2, }\DataTypeTok{type =} \StringTok{"cohort"}\NormalTok{)}
\CommentTok{#> Error in XenaBrowse(to_browse2, type = "cohort"): This function limite 1 cohort to browse. }
\CommentTok{#>  Set multiple to TRUE if you want to browse multiple links.}
\end{Highlighting}
\end{Shaded}

When you make sure you want to open multiple links, you can set
\texttt{multiple} option to \texttt{TRUE}.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{XenaBrowse}\NormalTok{(to_browse2, }\DataTypeTok{multiple =} \OtherTok{TRUE}\NormalTok{)}
\KeywordTok{XenaBrowse}\NormalTok{(to_browse2, }\DataTypeTok{type =} \StringTok{"cohort"}\NormalTok{, }\DataTypeTok{multiple =} \OtherTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{query}{%
\subsection{Query}\label{query}}

Create a query before downloading data.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{xe2_query =}\StringTok{ }\KeywordTok{XenaQuery}\NormalTok{(xe2)}
\CommentTok{#> This will check url status, please be patient.}
\NormalTok{xe2_query}
\CommentTok{#>                       hosts}
\CommentTok{#> 1 https://tcga.xenahubs.net}
\CommentTok{#> 2 https://tcga.xenahubs.net}
\CommentTok{#> 3 https://tcga.xenahubs.net}
\CommentTok{#>                                  datasets}
\CommentTok{#> 1 TCGA.LUAD.sampleMap/LUAD_clinicalMatrix}
\CommentTok{#> 2 TCGA.LUNG.sampleMap/LUNG_clinicalMatrix}
\CommentTok{#> 3 TCGA.LUSC.sampleMap/LUSC_clinicalMatrix}
\CommentTok{#>                                                                             url}
\CommentTok{#> 1 https://tcga.xenahubs.net/download/TCGA.LUAD.sampleMap/LUAD_clinicalMatrix.gz}
\CommentTok{#> 2 https://tcga.xenahubs.net/download/TCGA.LUNG.sampleMap/LUNG_clinicalMatrix.gz}
\CommentTok{#> 3 https://tcga.xenahubs.net/download/TCGA.LUSC.sampleMap/LUSC_clinicalMatrix.gz}
\end{Highlighting}
\end{Shaded}

\hypertarget{download}{%
\subsection{Download}\label{download}}

Default, data will be downloaded to system temp directory. You can
specify the path.

If the data exists, command will not run to download them, but you can
force it by \texttt{force} option.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{xe2_download =}\StringTok{ }\KeywordTok{XenaDownload}\NormalTok{(xe2_query)}
\CommentTok{#> All downloaded files will under directory /var/folders/mx/rfkl27z90c96wbmn3_kjk8c80000gn/T//RtmpWLoH68.}
\CommentTok{#> The 'trans_slash' option is FALSE, keep same directory structure as Xena.}
\CommentTok{#> Creating directories for datasets...}
\CommentTok{#> Downloading TCGA.LUAD.sampleMap/LUAD_clinicalMatrix.gz}
\CommentTok{#> Downloading TCGA.LUNG.sampleMap/LUNG_clinicalMatrix.gz}
\CommentTok{#> Downloading TCGA.LUSC.sampleMap/LUSC_clinicalMatrix.gz}
\end{Highlighting}
\end{Shaded}

Of note, at default, the downloaded files will keep same directory
structure as Xena. You can set \texttt{trans\_slash} to TRUE, it will
transform \texttt{/} in dataset id to \texttt{\_\_}, this will make all
downloaded files are under same directory.

\hypertarget{prepare}{%
\subsection{Prepare}\label{prepare}}

There are 4 ways to prepare data to R.

\begin{verbatim}
# way1:  directory
cli1 = XenaPrepare("E:/Github/XenaData/test/")
names(cli1)
## [1] "TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz"
## [2] "TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz"
## [3] "TCGA.LUSC.sampleMap__LUSC_clinicalMatrix.gz"
\end{verbatim}

\begin{verbatim}
# way2: local files
cli2 = XenaPrepare("E:/Github/XenaData/test/TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz")
class(cli2)
## [1] "tbl_df"     "tbl"        "data.frame"

cli2 = XenaPrepare(c("E:/Github/XenaData/test/TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz",
                     "E:/Github/XenaData/test/TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz"))
class(cli2)
## [1] "list"
names(cli2)
## [1] "TCGA.LUAD.sampleMap__LUAD_clinicalMatrix.gz"
## [2] "TCGA.LUNG.sampleMap__LUNG_clinicalMatrix.gz"
\end{verbatim}

\begin{verbatim}
# way3: urls
cli3 = XenaPrepare(xe2_download$url[1:2])
names(cli3)
## [1] "LUSC_clinicalMatrix.gz" "LUNG_clinicalMatrix.gz"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# way4: xenadownload object}
\NormalTok{cli4 =}\StringTok{ }\KeywordTok{XenaPrepare}\NormalTok{(xe2_download)}
\KeywordTok{names}\NormalTok{(cli4)}
\CommentTok{#> [1] "LUAD_clinicalMatrix.gz"}
\CommentTok{#> [2] "LUNG_clinicalMatrix.gz"}
\CommentTok{#> [3] "LUSC_clinicalMatrix.gz"}
\end{Highlighting}
\end{Shaded}

From v0.2.6, \texttt{XenaPrepare()} can enable chunk feature when file
is too big and user only need subset of file.

Following code show how to subset some rows or columns of files,
\texttt{sample} is the name of the first column, user can directly use
it in logical expression, \texttt{x} can be a representation of data
frame user wanna do subset operation. More custom operation can be set
as a function and pass to \texttt{callback} option.

\begin{verbatim}
# select rows which sample (gene symbol here) in "HIF3A" or "RNF17"
testRNA = UCSCXenaTools::XenaPrepare("~/Download/HiSeqV2.gz", use_chunk = TRUE, subset_rows = sample %in% c("HIF3A", "RNF17"))
# only keep 1 to 3 columns
testRNA = UCSCXenaTools::XenaPrepare("~/Download/HiSeqV2.gz", use_chunk = TRUE, select_cols = colnames(x)[1:3])
\end{verbatim}

\hypertarget{download-tcga-data-with-readable-options}{%
\section{Download TCGA data with readable
options}\label{download-tcga-data-with-readable-options}}

\hypertarget{gettcgadata}{%
\subsection{getTCGAdata}\label{gettcgadata}}

\texttt{getTCGAdata} provides a more readable way for downloading TCGA
(hg19 version, different from \texttt{gdcHub}) datasets, user can
specify multiple options to select data and corresponding file type to
download. Default this function will return a list include
\texttt{XenaHub} object and selected datasets information. Once you are
sure the datasets are exactly what you want, \texttt{download} can be
set to \texttt{TRUE} to download the data.

Check arguments of \texttt{getTCGAdata}:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{args}\NormalTok{(getTCGAdata)}
\CommentTok{#> function (project = NULL, clinical = TRUE, download = FALSE, }
\CommentTok{#>     forceDownload = FALSE, destdir = tempdir(), mRNASeq = FALSE, }
\CommentTok{#>     mRNAArray = FALSE, mRNASeqType = "normalized", miRNASeq = FALSE, }
\CommentTok{#>     exonRNASeq = FALSE, RPPAArray = FALSE, ReplicateBaseNormalization = FALSE, }
\CommentTok{#>     Methylation = FALSE, MethylationType = c("27K", "450K"), }
\CommentTok{#>     GeneMutation = FALSE, SomaticMutation = FALSE, GisticCopyNumber = FALSE, }
\CommentTok{#>     Gistic2Threshold = TRUE, CopyNumberSegment = FALSE, RemoveGermlineCNV = TRUE, }
\CommentTok{#>     ...) }
\CommentTok{#> NULL}

\CommentTok{# or run ??getTCGAdata to read documentation}
\end{Highlighting}
\end{Shaded}

Select one or more projects, default will select only clinical datasets:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{getTCGAdata}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\StringTok{"UVM"}\NormalTok{, }\StringTok{"LUAD"}\NormalTok{))}
\CommentTok{#> $Xena}
\CommentTok{#> class: XenaHub }
\CommentTok{#> hosts():}
\CommentTok{#>   https://tcga.xenahubs.net}
\CommentTok{#> cohorts() (2 total):}
\CommentTok{#>   TCGA Lung Adenocarcinoma (LUAD)}
\CommentTok{#>   TCGA Ocular melanomas (UVM)}
\CommentTok{#> datasets() (2 total):}
\CommentTok{#>   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix}
\CommentTok{#>   TCGA.UVM.sampleMap/UVM_clinicalMatrix}
\CommentTok{#> }
\CommentTok{#> $DataInfo}
\CommentTok{#> # A tibble: 2 x 20}
\CommentTok{#>   XenaHosts XenaHostNames XenaCohorts}
\CommentTok{#>   <chr>     <chr>         <chr>      }
\CommentTok{#> 1 https://~ tcgaHub       TCGA Lung ~}
\CommentTok{#> 2 https://~ tcgaHub       TCGA Ocula~}
\CommentTok{#> # ... with 17 more variables:}
\CommentTok{#> #   XenaDatasets <chr>, SampleCount <chr>,}
\CommentTok{#> #   DataSubtype <chr>, Label <chr>,}
\CommentTok{#> #   Type <chr>, AnatomicalOrigin <chr>,}
\CommentTok{#> #   SampleType <chr>, Tags <chr>,}
\CommentTok{#> #   ProbeMap <chr>, LongTitle <chr>,}
\CommentTok{#> #   Citation <chr>, Version <chr>,}
\CommentTok{#> #   Unit <chr>, Platform <chr>,}
\CommentTok{#> #   ProjectID <chr>, DataType <chr>,}
\CommentTok{#> #   FileType <chr>}

\NormalTok{tcga_data =}\StringTok{ }\KeywordTok{getTCGAdata}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\StringTok{"UVM"}\NormalTok{, }\StringTok{"LUAD"}\NormalTok{))}

\CommentTok{# only return XenaHub object}
\NormalTok{tcga_data}\OperatorTok{$}\NormalTok{Xena}
\CommentTok{#> class: XenaHub }
\CommentTok{#> hosts():}
\CommentTok{#>   https://tcga.xenahubs.net}
\CommentTok{#> cohorts() (2 total):}
\CommentTok{#>   TCGA Lung Adenocarcinoma (LUAD)}
\CommentTok{#>   TCGA Ocular melanomas (UVM)}
\CommentTok{#> datasets() (2 total):}
\CommentTok{#>   TCGA.LUAD.sampleMap/LUAD_clinicalMatrix}
\CommentTok{#>   TCGA.UVM.sampleMap/UVM_clinicalMatrix}

\CommentTok{# only return datasets information}
\NormalTok{tcga_data}\OperatorTok{$}\NormalTok{DataInfo}
\CommentTok{#> # A tibble: 2 x 20}
\CommentTok{#>   XenaHosts XenaHostNames XenaCohorts}
\CommentTok{#>   <chr>     <chr>         <chr>      }
\CommentTok{#> 1 https://~ tcgaHub       TCGA Lung ~}
\CommentTok{#> 2 https://~ tcgaHub       TCGA Ocula~}
\CommentTok{#> # ... with 17 more variables:}
\CommentTok{#> #   XenaDatasets <chr>, SampleCount <chr>,}
\CommentTok{#> #   DataSubtype <chr>, Label <chr>,}
\CommentTok{#> #   Type <chr>, AnatomicalOrigin <chr>,}
\CommentTok{#> #   SampleType <chr>, Tags <chr>,}
\CommentTok{#> #   ProbeMap <chr>, LongTitle <chr>,}
\CommentTok{#> #   Citation <chr>, Version <chr>,}
\CommentTok{#> #   Unit <chr>, Platform <chr>,}
\CommentTok{#> #   ProjectID <chr>, DataType <chr>,}
\CommentTok{#> #   FileType <chr>}
\end{Highlighting}
\end{Shaded}

Set \texttt{download=TRUE} to download data, default data will be
downloaded to system temp directory (you can specify the path with
\texttt{destdir} option):

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# only download clinical data}
\KeywordTok{getTCGAdata}\NormalTok{(}\KeywordTok{c}\NormalTok{(}\StringTok{"UVM"}\NormalTok{, }\StringTok{"LUAD"}\NormalTok{), }\DataTypeTok{download =} \OtherTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\textbf{Support Data Type and Options}:

\begin{itemize}
\tightlist
\item
  clinical information: \texttt{clinical}
\item
  mRNA Sequencing: \texttt{mRNASeq}
\item
  mRNA microarray: \texttt{mRNAArray}
\item
  miRNA Sequencing: \texttt{miRNASeq}
\item
  exon Sequencing: \texttt{exonRNASeq}
\item
  RPPA array: \texttt{RPPAArray}
\item
  DNA Methylation: \texttt{Methylation}
\item
  Gene mutation: \texttt{GeneMutation}
\item
  Somatic mutation: \texttt{SomaticMutation}
\item
  Gistic2 Copy Number: \texttt{GisticCopyNumber}
\item
  Copy Number Segment: \texttt{CopyNumberSegment}
\end{itemize}

\begin{quote}
other data type supported by Xena cannot download use this function.
Please refer to \texttt{downloadTCGA} function or \texttt{XenaGenerate}
function.
\end{quote}

\textbf{NOTE}: Sequencing data are all based on Illumina Hiseq platform,
other platform (Illumina GA) data supported by Xena cannot download
using this function. This is for building consistent data download flow.
Mutation use \texttt{broad\ automated} version (except \texttt{PANCAN}
use \texttt{MC3\ Public\ Version}). If you wan to download other
datasets, please refer to \texttt{downloadTCGA} function or
\texttt{XenaGenerate} function.

\hypertarget{download-any-tcga-data-by-datatypes-and-filetypes}{%
\subsection{Download any TCGA data by datatypes and
filetypes}\label{download-any-tcga-data-by-datatypes-and-filetypes}}

\texttt{downloadTCGA} function can be used to download any TCGA data
supported by Xena, but in a way different from \texttt{getTCGAdata}
function.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# download RNASeq data (use UVM as an example)}
\KeywordTok{downloadTCGA}\NormalTok{(}\DataTypeTok{project =} \StringTok{"UVM"}\NormalTok{, }\DataTypeTok{data_type =} \StringTok{"Gene Expression RNASeq"}\NormalTok{, }
    \DataTypeTok{file_type =} \StringTok{"IlluminaHiSeq RNASeqV2"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

See the arguments:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{args}\NormalTok{(downloadTCGA)}
\CommentTok{#> function (project = NULL, data_type = NULL, file_type = NULL, }
\CommentTok{#>     destdir = tempdir(), force = FALSE, ...) }
\CommentTok{#> NULL}
\end{Highlighting}
\end{Shaded}

Except \texttt{destdir} option, you only need to select three arguments
for downloading data. Even throught the number is far less than
\texttt{getTCGAdata}, it is more complex than the latter.

Before you download data, you need spare some time to figure out what
data type and file type available and what your datasets have.

\texttt{availTCGA} can return all information you need:

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{availTCGA}\NormalTok{()}
\CommentTok{#> Note not all projects have listed data types and file types, you can use showTCGA function to check if exist}
\CommentTok{#> $ProjectID}
\CommentTok{#>  [1] "LAML"     "ACC"      "CHOL"    }
\CommentTok{#>  [4] "BLCA"     "BRCA"     "CESC"    }
\CommentTok{#>  [7] "COADREAD" "COAD"     "UCEC"    }
\CommentTok{#> [10] "ESCA"     "FPPP"     "GBM"     }
\CommentTok{#> [13] "HNSC"     "KICH"     "KIRC"    }
\CommentTok{#> [16] "KIRP"     "DLBC"     "LIHC"    }
\CommentTok{#> [19] "LGG"      "GBMLGG"   "LUAD"    }
\CommentTok{#> [22] "LUNG"     "LUSC"     "SKCM"    }
\CommentTok{#> [25] "MESO"     "UVM"      "OV"      }
\CommentTok{#> [28] "PANCAN"   "PAAD"     "PCPG"    }
\CommentTok{#> [31] "PRAD"     "READ"     "SARC"    }
\CommentTok{#> [34] "STAD"     "TGCT"     "THYM"    }
\CommentTok{#> [37] "THCA"     "UCS"     }
\CommentTok{#> }
\CommentTok{#> $DataType}
\CommentTok{#>  [1] "DNA Methylation"                       }
\CommentTok{#>  [2] "Gene Level Copy Number"                }
\CommentTok{#>  [3] "Somatic Mutation"                      }
\CommentTok{#>  [4] "Gene Expression RNASeq"                }
\CommentTok{#>  [5] "miRNA Mature Strand Expression RNASeq" }
\CommentTok{#>  [6] "Gene Somatic Non-silent Mutation"      }
\CommentTok{#>  [7] "Copy Number Segments"                  }
\CommentTok{#>  [8] "Exon Expression RNASeq"                }
\CommentTok{#>  [9] "Phenotype"                             }
\CommentTok{#> [10] "PARADIGM Pathway Activity"             }
\CommentTok{#> [11] "Protein Expression RPPA"               }
\CommentTok{#> [12] "Transcription Factor Regulatory Impact"}
\CommentTok{#> [13] "Gene Expression Array"                 }
\CommentTok{#> [14] "Signatures"                            }
\CommentTok{#> [15] "iCluster"                              }
\CommentTok{#> }
\CommentTok{#> $FileType}
\CommentTok{#>  [1] "Methylation27K"                            }
\CommentTok{#>  [2] "Methylation450K"                           }
\CommentTok{#>  [3] "Gistic2"                                   }
\CommentTok{#>  [4] "wustl hiseq automated"                     }
\CommentTok{#>  [5] "IlluminaGA RNASeq"                         }
\CommentTok{#>  [6] "IlluminaHiSeq RNASeqV2 in percentile rank" }
\CommentTok{#>  [7] "IlluminaHiSeq RNASeqV2 pancan normalized"  }
\CommentTok{#>  [8] "IlluminaHiSeq RNASeqV2"                    }
\CommentTok{#>  [9] "After remove germline cnv"                 }
\CommentTok{#> [10] "PANCAN AWG analyzed"                       }
\CommentTok{#> [11] "Clinical Information"                      }
\CommentTok{#> [12] "wustl automated"                           }
\CommentTok{#> [13] "Gistic2 thresholded"                       }
\CommentTok{#> [14] "Before remove germline cnv"                }
\CommentTok{#> [15] "Use only RNASeq"                           }
\CommentTok{#> [16] "Use RNASeq plus Copy Number"               }
\CommentTok{#> [17] "bcm automated"                             }
\CommentTok{#> [18] "IlluminaHiSeq RNASeq"                      }
\CommentTok{#> [19] "bcm curated"                               }
\CommentTok{#> [20] "broad curated"                             }
\CommentTok{#> [21] "RPPA"                                      }
\CommentTok{#> [22] "bsgsc automated"                           }
\CommentTok{#> [23] "broad automated"                           }
\CommentTok{#> [24] "bcgsc automated"                           }
\CommentTok{#> [25] "ucsc automated"                            }
\CommentTok{#> [26] "RABIT Use IlluminaHiSeq RNASeqV2"          }
\CommentTok{#> [27] "RABIT Use IlluminaHiSeq RNASeq"            }
\CommentTok{#> [28] "RPPA normalized by RBN"                    }
\CommentTok{#> [29] "RABIT Use Agilent 244K Microarray"         }
\CommentTok{#> [30] "wustl curated"                             }
\CommentTok{#> [31] "Use Microarray plus Copy Number"           }
\CommentTok{#> [32] "Use only Microarray"                       }
\CommentTok{#> [33] "Agilent 244K Microarray"                   }
\CommentTok{#> [34] "IlluminaGA RNASeqV2"                       }
\CommentTok{#> [35] "bcm SOLiD"                                 }
\CommentTok{#> [36] "RABIT Use IlluminaGA RNASeqV2"             }
\CommentTok{#> [37] "RABIT Use IlluminaGA RNASeq"               }
\CommentTok{#> [38] "RABIT Use Affymetrix U133A Microarray"     }
\CommentTok{#> [39] "Affymetrix U133A Microarray"               }
\CommentTok{#> [40] "MethylMix"                                 }
\CommentTok{#> [41] "bcm SOLiD curated"                         }
\CommentTok{#> [42] "Gene Expression Subtype"                   }
\CommentTok{#> [43] "Platform-corrected PANCAN12 dataset"       }
\CommentTok{#> [44] "Batch effects normalized"                  }
\CommentTok{#> [45] "MC3 Public Version"                        }
\CommentTok{#> [46] "TCGA Sample Type and Primary Disease"      }
\CommentTok{#> [47] "RPPA pancan normalized"                    }
\CommentTok{#> [48] "Tumor copy number"                         }
\CommentTok{#> [49] "Genome-wide DNA Damage Footprint HRD Score"}
\CommentTok{#> [50] "TCGA Molecular Subtype"                    }
\CommentTok{#> [51] "iCluster cluster assignments"              }
\CommentTok{#> [52] "iCluster latent variables"                 }
\CommentTok{#> [53] "RNA based StemnessScore"                   }
\CommentTok{#> [54] "DNA methylation based StemnessScore"       }
\CommentTok{#> [55] "Pancan Gene Programs"                      }
\CommentTok{#> [56] "Immune Model Based Subtype"                }
\CommentTok{#> [57] "Immune Signature Scores"}
\end{Highlighting}
\end{Shaded}

Note not all datasets have these property, \texttt{showTCGA} can help
you to check it. It will return all data in TCGA, you can use following
code in RStudio and search your data.

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{View}\NormalTok{(}\KeywordTok{showTCGA}\NormalTok{())}
\end{Highlighting}
\end{Shaded}

\textbf{OR you can use shiny app provided by \texttt{UCSCXenaTools} to
search}.

Run shiny by:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{UCSCXenaTools}\OperatorTok{::}\KeywordTok{XenaShiny}\NormalTok{()}
\end{Highlighting}
\end{Shaded}

\hypertarget{sessioninfo}{%
\section{SessionInfo}\label{sessioninfo}}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{sessionInfo}\NormalTok{()}
\CommentTok{#> R version 3.6.0 RC (2019-04-21 r76409)}
\CommentTok{#> Platform: x86_64-apple-darwin15.6.0 (64-bit)}
\CommentTok{#> Running under: macOS High Sierra 10.13.6}
\CommentTok{#> }
\CommentTok{#> Matrix products: default}
\CommentTok{#> BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib}
\CommentTok{#> LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib}
\CommentTok{#> }
\CommentTok{#> locale:}
\CommentTok{#> [1] zh_CN.UTF-8/zh_CN.UTF-8/zh_CN.UTF-8/C/zh_CN.UTF-8/zh_CN.UTF-8}
\CommentTok{#> }
\CommentTok{#> attached base packages:}
\CommentTok{#> [1] stats     graphics  grDevices utils    }
\CommentTok{#> [5] datasets  methods   base     }
\CommentTok{#> }
\CommentTok{#> other attached packages:}
\CommentTok{#> [1] dplyr_0.8.1         UCSCXenaTools_1.2.2}
\CommentTok{#> [3] pacman_0.5.1       }
\CommentTok{#> }
\CommentTok{#> loaded via a namespace (and not attached):}
\CommentTok{#>  [1] Rcpp_1.0.1          }
\CommentTok{#>  [2] pillar_1.4.1        }
\CommentTok{#>  [3] compiler_3.6.0      }
\CommentTok{#>  [4] formatR_1.7         }
\CommentTok{#>  [5] later_0.8.0         }
\CommentTok{#>  [6] tools_3.6.0         }
\CommentTok{#>  [7] zeallot_0.1.0       }
\CommentTok{#>  [8] digest_0.6.19       }
\CommentTok{#>  [9] evaluate_0.14       }
\CommentTok{#> [10] tibble_2.1.3        }
\CommentTok{#> [11] pkgconfig_2.0.2     }
\CommentTok{#> [12] rlang_0.3.4         }
\CommentTok{#> [13] cli_1.1.0           }
\CommentTok{#> [14] shiny_1.3.2         }
\CommentTok{#> [15] curl_3.3            }
\CommentTok{#> [16] yaml_2.2.0          }
\CommentTok{#> [17] xfun_0.7            }
\CommentTok{#> [18] stringr_1.4.0       }
\CommentTok{#> [19] httr_1.4.0          }
\CommentTok{#> [20] knitr_1.23          }
\CommentTok{#> [21] vctrs_0.1.0         }
\CommentTok{#> [22] hms_0.4.2           }
\CommentTok{#> [23] shinydashboard_0.7.1}
\CommentTok{#> [24] tidyselect_0.2.5    }
\CommentTok{#> [25] glue_1.3.1          }
\CommentTok{#> [26] R6_2.4.0            }
\CommentTok{#> [27] fansi_0.4.0         }
\CommentTok{#> [28] rmarkdown_1.13      }
\CommentTok{#> [29] purrr_0.3.2         }
\CommentTok{#> [30] readr_1.3.1         }
\CommentTok{#> [31] magrittr_1.5        }
\CommentTok{#> [32] backports_1.1.4     }
\CommentTok{#> [33] promises_1.0.1      }
\CommentTok{#> [34] htmltools_0.3.6     }
\CommentTok{#> [35] assertthat_0.2.1    }
\CommentTok{#> [36] tint_0.1.2          }
\CommentTok{#> [37] mime_0.7            }
\CommentTok{#> [38] xtable_1.8-4        }
\CommentTok{#> [39] httpuv_1.5.1        }
\CommentTok{#> [40] utf8_1.1.4          }
\CommentTok{#> [41] stringi_1.4.3       }
\CommentTok{#> [42] crayon_1.3.4}
\end{Highlighting}
\end{Shaded}

\hypertarget{bug-report}{%
\section{Bug Report}\label{bug-report}}

I have no time to test if all conditions are right and all datasets can
normally be downloaded. So if you have any question or suggestion,
please open an issue on Github at
\url{https://github.com/ShixiangWang/UCSCXenaTools/issues}.

\hypertarget{acknowledgement}{%
\section{Acknowledgement}\label{acknowledgement}}

This package is based on
\href{https://github.com/mtmorgan/XenaR}{XenaR}, thanks
\href{https://github.com/mtmorgan}{Martin Morgan} for his work.

\hypertarget{license}{%
\section{LICENSE}\label{license}}

GPL-3

Please note, code from XenaR package under Apache 2.0 license.



\end{document}
